elements/lisk-utils/test/math/q.spec.ts

```diff
@@ -0,0 +1,325 @@
+/*
+ * Copyright © 2022 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+
+import { Q_OPERATION, q, q96 } from '../../src/math';
+
+/**
+ * For the test cases `binary` represents the calculation result in binary.
+ */
+describe('Q', () => {
+	//  In this test Q8 is used as example because 8 bit is easier to generate the tests.
+	// For Q8 after 9th bit should be ignored
+	const base = 8;
+	const pow2 = (val: number | bigint) => BigInt(2) ** BigInt(val);
+	// mulBasePow2 computes 2 ** base * val
+	const mulBasePow2 = (b: number, val: number | bigint) => pow2(b) * BigInt(val);
+	// pow2BaseDiffAddition accumulates 2 ** (base-vals[n])
+	const pow2BaseDiffAddition = (b: number, vals: number[]) =>
+		vals.reduce((prev, curr) => prev + pow2(b - curr), BigInt(0));
+	// ExpectedQ generates the Q format big integer
+	// int: integer part of the result
+	// decimalBinaryPosthe elements should bein the array means the position of "1" in binary.
+	const expectedQ = (b: number, int: number, decimalBinaryPos?: number[]) =>
+		mulBasePow2(b, int) + pow2BaseDiffAddition(b, decimalBinaryPos ?? []);
```

Comment:
Can we get rid of all this junk by using cleaner test data? 🙂

Reply:
Any suggestion to construct good test data?

Reply:
Cleaned up and pushed 🧹